src/db/couchDevices.ts

```diff
@@ -46,6 +47,19 @@ export const asCouchDevice = asCouchDoc(
     deviceToken: asOptional(asString),
     ignoreMarketing: asOptional(asBoolean, false),
     ignorePriceChanges: asOptional(asBoolean, false),
+    ip: asOptional(asString, 'unknown'),
```

Comment:
Can we just allow the ip to be `undefined` if missing? If we ever try parsing the string "unknown" as an IP address, it will certainly throw, whereas a mandatory `if (row.ip != null)` gets the type system on our side.

Reply:
I meant to make `ip === 'unknown'` distinct from `'ip === undefined'` in order to know if a particular document has migrated or not. But this isn't too important, so I'll go with your suggestion.